ClientApp/viking-vault/src/display_user_list/UserData.tsx

```diff
@@ -24,59 +40,123 @@ const defaultUser = {
+                addMoneyResponseMessage: ""
             };
         }
         else
         {
-            this.state = {...defaultUser};
+            this.state = {
+                        user : {...defaultUser},
+                        openDeleteUserModal : false,
+                        openAddMoneyFormModal : false,
+                        openAddMoneyResponseModal: false,
+                        addMoneyResponseMessage: ""
+                    };
         }        
     }
 
     renderUserCard()
     {
-        if(this.state.cardNumber !== "")
+        if(this.state.user.cardNumber !== "")
             return <div className = "card-data-style">
-                        <p className = "name-on-card">{this.state.firstName} {this.state.lastName}</p>
-                        <p className = "expiration-date-on-card">{this.state.expirationDate}</p>
-                        <p className = "card-number-on-card">{this.state.cardNumber}</p>
+                        <p className = "name-on-card">{this.state.user.firstName} {this.state.user.lastName}</p>
+                        <p className = "expiration-date-on-card">{this.state.user.expirationDate}</p>
+                        <p className = "card-number-on-card">{this.state.user.cardNumber}</p>
                         <img src = {CardImg} className = "card-img"></img>
                    </div>        
 
         return null; 
     }
+    
+    private handleAddMoney = () =>{
+        this.setState((oldstate : any)=>({
+            openAddMoneyFormModal : !oldstate.openAddMoneyFormModal
+        }));
+    }
+
+    private closeAddMoneyFormModal = () =>{
+        this.setState({
+            openAddMoneyFormModal : false
+        });
+    }
+
+    private closeAddMoneyResponseModal = () =>{
+        this.setState({
+            openAddMoneyResponseModal : false,
+        });
+    }
+
+    private openResponseModalWithMessage = (message : string) =>{
+        this.setState({
+            openAddMoneyResponseModal : true,
+            addMoneyResponseMessage : message
+        });
+    }
+
+    private addMoney = (amount : Number) =>{
+        this.setState({
+            openAddMoneyFormModal : false
+        });
+        fetch(constants.baseUrl+"bankAccount", {
+            method: "PUT",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+              CurrencyType: "Ron",
+              Amount: amount,
+              Email: this.state.user.email
+            })}).then(response => {
+                if(response.status === 200) {
+                    this.openResponseModalWithMessage("Added RON "+amount+" to "+this.state.user.firstName + " "+this.state.user.lastName+" account!");
+                }
+                
+                if(response.status !== 200) {
+                    this.openResponseModalWithMessage("Something wrong happened. Try again later!");
```

Comment:
You can also append a `.catch` after then last then to check if the request itself failed. Right now you only display the error message if the response failed, not the request itself.